src/bindings/python/flux/accounting/accounting_cli_functions.py

```diff
@@ -66,15 +66,56 @@ def view_bank(conn, bank):
 
 
 def delete_bank(conn, bank):
-    last_parent_bank_seen = bank
-
-    # delete bank from bank_table
-    delete_stmt = "DELETE FROM bank_table WHERE bank=?"
-    cursor = conn.cursor()
-    cursor.execute(delete_stmt, (bank,))
+    try:
+        # delete the first bank passed in
+        delete_stmt = """
+            DELETE FROM bank_table
+            WHERE bank=?
+            """
+        cursor = conn.cursor()
+        cursor.execute(delete_stmt, (bank,))
+
+        # construct a DataFrame object out of the
+        # bank passed in
+        df = pd.DataFrame([bank], columns=["bank"])
+        bank = df.iloc[0]
+
+        # helper function to traverse the bank table
+        # and delete all of its sub banks
+        def get_sub_banks(row):
+            select_stmt = """
+                SELECT bank
+                FROM bank_table
+                WHERE parent_bank=?
+                """
+            cursor = conn.cursor()
+            dataframe = pd.read_sql_query(select_stmt, conn, params=(row["bank"],))
+
+            # we've reached a bank with no sub banks
+            if len(dataframe) == 0:
+                select_associations_stmt = """
+                    SELECT user_name, bank
+                    FROM association_table
+                    WHERE bank=?
+                    """
+                for row in cursor.execute(select_associations_stmt, (row["bank"],)):
+                    delete_user(conn, user=row[0], bank=row[1])
+            # else, delete all of its sub banks and continue
+            # traversing
+            else:
+                for index, sub_bank_row in dataframe.iterrows():
+                    cursor.execute(delete_stmt, (sub_bank_row["bank"],))
+                    get_sub_banks(sub_bank_row)
+
+        get_sub_banks(bank)
 
-    # commit changes
-    conn.commit()
+        conn.commit()
+    # if an exception occcurs while recursively deleting
+    # the parent banks, then throw the exception and roll
+    # back the changes made to the DB
+    except sqlite3.OperationalError as e:
+        print(e)
+        conn.execute("ROLLBACK")
```

Comment:
It looks like python's `sqlite3` module has built-in support for rollback: https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.rollback

```suggestion
        conn.rollback()
```